### PR TITLE
External Transaction Signing

### PR DESCRIPTION
**Pull request by shemnon**
An RPC protocol is described that would allow clients and SDKs to sign
transactions with external programs, devices, and services.

### CODE REVIEW

File: HIP/hip-external-transaction-signing.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+services with actual people gate-keeping activity. A process needs to be adopted
+that can seamlessly support both developer grade security and full production
+security.
+
+## Rationale
+
+As part of a larger DLT ecosystem Hedera should be better served by adopting
+existing cross-chain standards. Hence, the utility offered by teh Chain Agnostic
+Improvement Proposal group relating to RPC interactions was chosen as the main
+interface. In particular CAIP-25 and CAIP-27 provide the bulk of hte needed
```
Comment (sergmetelin) [inline HIP/hip-external-transaction-signing.md:37]: Typo here: `hte` -> `the`

File: HIP/hip-external-transaction-signing.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+
+## Abstract
+
+An RPC protocol is described that would allow clients and SDKs to sign
+transactions with external programs, devices, and services.
+
+## Motivation
+
+The current Hedera SDK requires that the private keys for accounts be available
+and signed in process. One process is to provide the private key in an
```
Comment (bugbytesinc) [inline HIP/hip-external-transaction-signing.md:22]: Can this be a bit more specific? the HAPI does not require private keys, and some of the sdks already provide mechanisms to sign transactions without receiving a private key.

File: HIP/hip-external-transaction-signing.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+### Hedera signing JSON-RPC calls
+
+Two JSON-RPC methods are defined for use in the CAIP-27 provider
+request, `hedera_signTransaction` and `hedera_sendTransaction`.
+
+#### `hedera_signTransaction`
+
+This JSON-RPC method signs a transaction that can be submitted to the network
+via the exiting SDK methods. It is expected there will be some operation on the
```
Comment (sergmetelin) [inline HIP/hip-external-transaction-signing.md:87]: `exiting` -> `existing`

**Comment by shemnon** [comment]
The protocol in this rendition doesn't support signing multiple keys in one call.  Since the signatures are independent (i.e. on sig does not sign the tx with other sigs) multiple parallel sign requests via `sign_transaction` with different keys can be sent and then aggregated before the transaction is sent if you manually aggregate and send.  Only one key is supported for `send_transaction`.

File: HIP/hip-external-transaction-signing.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+real-world value and the needed security posture needs to be better than "good."
+Typical high security setups involve hardware devices storing keys or external
+services with actual people gate-keeping activity. A process needs to be adopted
+that can seamlessly support both developer grade security and full production
+security.
+
+## Rationale
+
+As part of a larger DLT ecosystem Hedera should be better served by adopting
+existing cross-chain standards. Hence, the utility offered by teh Chain Agnostic
```
Comment (brianjohnson5972) [inline HIP/hip-external-transaction-signing.md:35]: teh ==> the

File: HIP/hip-external-transaction-signing.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+Improvement Proposal group relating to RPC interactions was chosen as the main
+interface. In particular CAIP-25 and CAIP-27 provide the bulk of hte needed
+protocol. Some additional definitions of hedera signing operations is needed to
+finish our the full loop.
```
Comment (brianjohnson5972) [inline HIP/hip-external-transaction-signing.md:39]: "our the"? I think you want either just "the" or "our"

File: HIP/hip-external-transaction-signing.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+>     "id": 1,
+>     "jsonrpc": "2.0",
+>     "method": "hedera_signTransaction",
+>     "params": {
+>         "transaction": "fedbca9876543210"
+>     }
+> }
+> ```
+
+> An example specifying a private key
```
Comment (sergmetelin) [inline HIP/hip-external-transaction-signing.md:109]: Probably safer to say you specify a `public` key here, not `private`

File: HIP/hip-external-transaction-signing.md (hunk 1 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: done

File: HIP/hip-external-transaction-signing.md (hunk 2 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: Only the env methods are documented.  While they provide hooks there are no default implementations.  The goal of this HIP is to start standardizing how those implementations might work.

File: HIP/hip-external-transaction-signing.md (hunk 3 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: done

File: HIP/hip-external-transaction-signing.md (hunk 4 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: done

File: HIP/hip-external-transaction-signing.md (hunk 5 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: just 'the' is what I went with.

File: HIP/hip-external-transaction-signing.md (hunk 6 above)
Comment (shemnon) [reply HIP/hip-external-transaction-signing.md]: done

**Review by sergmetelin — APPROVED** [review]
LGTM

**Pull request merged**